test/integration/scheduler/util.go

```diff
@@ -511,3 +512,38 @@ func createNamespacesWithLabels(cs clientset.Interface, namespaces []string, lab
 	}
 	return nil
 }
+
+// timeout returns a timeout error if the given `f` function doesn't
+// complete within `d` duration; otherwise it returns nil.
+func timeout(ctx context.Context, d time.Duration, f func()) error {
+	ctx, cancel := context.WithTimeout(ctx, d)
+	defer cancel()
+
+	done := make(chan struct{})
+	go func() {
+		f()
+		done <- struct{}{}
+	}()
+
+	select {
+	case <-done:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+// nextPodOrDie returns the next Pod in the scheduler queue.
+// The operation needs to be completed within 15 seconds; otherwise the test gets aborted.
+func nextPodOrDie(t *testing.T, testCtx *testutils.TestContext) *framework.QueuedPodInfo {
```

Comment:
we lost the flexibility to config how long the test case should abort, it's hard coded as 15 seconds now, is that necessary to pass another var here?

Reply:
Let's see if we have this requirement in the future. If so, we can easily compose another `nextPodOrDieWithTimeout` function.

Reply:
Add `t.Helper`

Reply:
done.